src/compiler/core.ts

```diff
@@ -2593,6 +2593,16 @@ namespace ts {
         return find<Extension>(supportedTypescriptExtensionsForExtractExtension, e => fileExtensionIs(path, e)) || find(supportedJavascriptExtensions, e => fileExtensionIs(path, e));
     }
 
+    // Retrieves any string from the final "." onwards from a base file name.
+    // Unlike extensionFromPath, which throws an exception on unrecognized extensions.
+    export function getAnyExtensionFromPath(path: string): string | undefined {
+        const baseFileName = getBaseFileName(path);
+        const extensionIndex = baseFileName.lastIndexOf(".");
+        if (extensionIndex >= 0) {

```

Comment:
What's the file extension of something like `.gitconfig`?  If we decide that it has no file extension (because that's not really what dot means in that position) we just have to change `>=` to `>`.

Reply:
I always think of those as files with just an extension. Not sure it matters in this case (e.g. if you have a file named `.html`, you probably want it treated as an HTML file.